File: Cpp/1346_Check_If_N_and_Its_Double_Exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_set<int> s;
        for(auto& num : arr){
            if((num % 2 == 0 && s.find(num/2) != s.end()) || s.find(num*2) != s.end()) return true;
            s.insert(num);
        }
        return false;
    }
};

File: Kotlin/1346_Check_If_N_and_Its_Double_Exist.kt
class Solution {
    fun checkIfExist(arr: IntArray): Boolean {
        val s = mutableSetOf<Int>()
        for(i in arr){
            if((i % 2 == 0 && s.contains(i / 2)) || s.contains(i * 2)) return true
            s.add(i)
        }
        
        return false
    }
}